scripts/c-to-cpp-includes.pl

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/perl
+
+# sweep over the squid repository, looking for patterns of
+# c standard library header files that have a c++ equivalent,
+# replacing these includes with the sequivalent standard c++ header.
+
+
+#header files to consider (without the .h extension)
+@headers=qw(
+assert
+ctype
+errno
+inttypes
+limits
+math
+signal
+stdarg
+stddef
+stdint
+stdio
+stdlib
+string
+time
+unistd
+);
+
+foreach $header (@headers) {
+    $headerguard="HAVE_".uc($header)."_H";
+    print "headerguard: $headerguard\n";
+    @files=`git grep -lF $headerguard -- *.cc *.h *.cci| grep -v -e ^compat`;
+    chomp(@files);
+    foreach $file (@files) { 
+        print "file: $file\n";
+        $pre="";
+        open(FILE,"<", $file);
+        read(FILE, $pre ,1000000) || die("read error on file $file: $!");
+        close(FILE) || die("close error on file $file: $!");
+        $post = $pre;
+        $post =~ s/#if(def)? $headerguard\n#include <$header.h>(.*)\n#endif/#include <c$header>$2/m;
```

Comment:
It will probably take us a while to agree what this script should do and then adjust it to do it well. I suggest dropping this script from the PR and just merging the (polished) changes.

If we want to flag policy violations (after applying these changes), we can add a (much simpler!) check that will _detect_ them. It would be easy for the developer to fix an occasional violation manually. Those occasional simple fixes are not worth the automation overheads in this case IMO (beyond using a script for creation of the initial set of changes, of course).

Reply:
That is actually the reason why I put it in scripts/ rather than in maintenance/; I expect this to be a one-off and that regular code reviews will prevent creep, while preserving the work done to write the automation

Reply:
> I expect this to be a one-off 

Please do not add one-off scripts to official repositories, especially one-off scripts that we would, in that case, need to spend time on (to agree on  the specifics and to polish). Official repositories are for scripts we intend to be reused quite a bit. This is not the only admission criteria, of course, but it is a critical one.

Reply:
> > I expect this to be a one-off
> 
> Please do not add one-off scripts to official repositories, especially one-off scripts that we would, in that case, need to spend time on (to agree on the specifics and to polish). Official repositories are for scripts we intend to be reused quite a bit. This is not the only admission criteria, of course, but it is a critical one.

I disagree on the philosophy, but the point is moot because the output of the script is the relevant part here so I will not discuss it